### ISSUE TITLE
Bug 1271974 - One or two "Highlights" tiles are not displayed after importing history from Chrome profile

### ISSUE THREAD

**Issue by SoftVision-PaulOiegas** [mozilla/activity-stream#675]
via https://bugzilla.mozilla.org/show_bug.cgi?id=1271974

<hr>

[Affected versions]:
- Firefox 45.0 and up
- Activity Stream 1.0.16

[Affected Platforms]:
- All Windows
- All Linux
- All Mac OS

[Notes]:
- Attached a Chrome profile data as history dataset to reproduce the issue. I have added in the attachment the steps for how to create the Chrome profile with the attached data (https://goo.gl/Fld5F5). 
- I have reproduced this issue with several different Chrome profiles, the one attached was a new Chrome profile on which I created some history in the last 2-3 days.
- Attached a screen recording of the issue https://goo.gl/9FK9xB.
- The issue is reproducible on the New Tab page as well. 

[Prerequisites]:
- Latest Activity Stream add-on version (1.0.16) was previously installed on a clean profile.  

[Steps to reproduce]:
1. Open the browser with the profile from prerequisites.
2. Click the Activity Stream button from the browser toolbar.
3. Press the "Ctrl+Shift+B" key combination.
4. After the "Library" window is opened, click "Import and Backup" button form the top of the window.
5. From the displayed context menu, click on the "Import data from another browser..." button.
6. Choose to import history from Chrome and select the profile from the provided attachment. 
7. Wait until the history is imported and after close the "Library" window.
8. Refresh the Timeline page.

[Expected result]:
- All 3 "Highlights" tiles are correctly populated with history from Chrome profile.

[Actual result]:
- Only two of the 3 tiles are displayed and populated with history from Chrome profile.

[Additional Notes]:
- If the all three "Highlights" tiles are displayed, remove the first "Highlight" tile from history. This will leave the "Timeline" with just 2 of the 3 tiles displayed.

**Comment by rlr** [comment]
Turns out we have issues when multiple sites/urls share the exact same `lastVisitDate` since we use that for the react key. I see a bunch of warnings in the console with the data provided. Like: 

```
Warning: flattenChildren(...): Encountered two children with the same key, `.$1462892403193`. Child keys must be unique; when two children share a key, only the first child will be used.
```

We need to change the key to something else. I'll propose something in a PR.

**Comment by rlr** [comment]
landed fix ^

**Comment by CosminMuntean** [comment]
This issue was retested on latest Nightly (49.0a1) with Activity Stream v1.0.17 and it is still reproducible.

A Chrome profile which has several visited websites was imported in Firefox. After the import, in History section from Firefox we can observe that there are some duplicate websites (eg. "bbc.com" and "www.bbc.com"). In Activity Stream v1.0.17 the duplicated websites are displayed in "Timeline" (they are overlaid in the same tile). If the duplicate items reach "Highlights", only 2 of the 3 tiles will be displayed. When a duplicate website is removed from "Highlights", the 3rd tile will be displayed.

In Activity Stream v1.0.16 even if in History section are duplicate websites after the import, in "Timeline" just one of them was displayed and all 3 "Highlights" tiles were correctly populated.

[Here](https://goo.gl/HWtG5u) is the link for a zip file that contains the Chrome profile used.

![activity stream timeline](https://cloud.githubusercontent.com/assets/19226534/15430375/caec0d14-1ead-11e6-8e7b-181643b968ab.png)
<hr>
This is how history looks in Chrome

![chrome profile](https://cloud.githubusercontent.com/assets/19226534/15430413/f5ec2850-1ead-11e6-8dd2-c8fb9ab70c79.png)

**Comment by rlr** [comment]
The history sidebar shows everything duplicate too. So we are showing what's in the Firefox history. The bug seems to be with the importing. Or did you import twice?

**Comment by rlr** [comment]
Yeah, if I import that profile into a fresh copy of Firefox, everything comes in duplicate into the Firefox history. That should be a bug against the Firefox importer.

<img width="797" alt="screen shot 2016-05-20 at 3 02 46 pm" src="https://cloud.githubusercontent.com/assets/36629/15438848/139e9886-1e9c-11e6-9768-c08ab45e07e9.png">

**Comment by SoftVision-CosminMuntean** [comment]
The websites were visited a single time in Chrome and the History was imported once. It seems that in the latest version of Activity Stream (v1.0.17-dev) all 3 "Highlights" tiles are populated with history, the empty tile is no longer displayed. However, we can observe that there are websites displayed twice in "Highlights".

After further investigation we have compared "about:newtab" with Activity Stream Timeline and observed the following:
- in "about:newtab", the websites are displayed only once, even if in History they are displayed multiple times after the History import from Chrome.
- in Activity Stream Timeline, the websites imported from the Chrome profile are displayed twice. The difference between several websites is the "www" part, like: bbc.com vs www.bbc.com
![duplicate websites](https://cloud.githubusercontent.com/assets/19226534/15472466/f331bf3c-2103-11e6-8615-88b71a038dbf.png)

**Comment by sarracini** [comment]
Closing this as a won't fix for now
